File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    sum:=0
    start:=0
    ans:= len(nums)+1
    for i:=0;i<len(nums);i++ {
        sum = sum + nums[i]
        // fmt.Println(sum, start, i, ans)
        for sum>=target {
            if ans > i - start + 1 {
                ans = i-start + 1
            }
            sum = sum - nums[start]
            start = start + 1
        }
    }
    return ans% (len(nums)+1)
    
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i,value = 0,0
        ans = float('inf')
        for j in range(len(nums)):
            value+=nums[j]
            while value>=target:
                ans = min(ans, j-i+1)
                value-= nums[i]
                i+=1
        return ans if ans!= float('inf') else 0
